meson.build

```diff
@@ -1119,32 +1119,16 @@ libfdisk = dependency('fdisk',
                       required : get_option('fdisk'))
 conf.set10('HAVE_LIBFDISK', libfdisk.found())
 
-want_passwdqc = get_option('passwdqc')
-want_pwquality = get_option('pwquality')
-if want_passwdqc == 'true' and want_pwquality == 'true'
-        error('passwdqc and pwquality cannot be requested simultaneously')
-endif
-
-if want_pwquality != 'false' and want_passwdqc != 'true' and not skip_deps
-        libpwquality = dependency('pwquality',
-                                  version : '>= 1.4.1',
-                                  required : want_pwquality == 'true')
-        have = libpwquality.found()
-else
-        have = false
-        libpwquality = []
+libpwquality = dependency('pwquality',
+                          version : '>= 1.4.1',
+                          required : get_option('pwquality'))
+have = libpwquality.found()
+if not have
+        libpwquality = dependency('passwdqc',
+                                  required : get_option('passwdqc'))
```

Comment:
This is problematic, as even if pwquality is auto and passwdqc is enabled, passwdqc will not be used when pwquality is installed.

Reply:
I don't see an issue with that. If you request one, you should be disabling the other. In fact, the old code made requesting both an error so this question doesn't arise.

Reply:
I just realized that this is the use case for `disable_auto_if()`. I've changed it to use that so that `-Dpwquality=auto -Dpasswdqc=enabled` should pick the latter, while `-Dpwquality=enabled -Dpasswdqc=enabled` will prefer the former. Same with libidn.